### PR TITLE
Truck Inspection Plugin

### PR DESCRIPTION
**Pull request by qswawrq**
Initial commit for Truck Inspection Plugin, which contains logic to work with UI and sends out truck safety log request.
Implement https://github.com/usdot-fhwa-stol/CARMAPlatform/issues/613

### CODE REVIEW

**Comment by MishkaMN** [comment]
Looks good to me

**Comment by maefromm** [comment]
@qswawrq as discussed with Dan, the following needs to be accomodated: 

1. "cav_truck_identified" topic needs to be a String not Bool, to include the VIN per John's initial slide presentation and what I noted on my review of your design doc.

2. "truck_safety_info" needs to be republishing the same information for the requested VIN every second for easy troublshooting during integation testing and in case the message falls through. 

3. "truck_safety_info" needs to include the receiveddate for the MobilityOperation Header in order for the UI to determine that a new safety info has been received. 

4. "send_truck_inspection_request" when this is sent again by the UI for the same VIN, please make sure it stops publishing (if possible). We will be comparing the date on the UI but would be best to have the topic not publish old information when a new request has been sent and you're still waiting for new MobilityOperation to come in.

File: truck_inspection_plugin/src/truck_inspection_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            // if the incoming message contains a valid safety log
+            if(isSafetyLogValid(msg->strategy_params))
+            {
+                safety_log_ = msg->strategy_params;
+            } else {
+                std_msgs::String msg_out;
+                std::string k_v_pair = msg->strategy_params;
+                // get only VIN number
+                msg_out.data = k_v_pair.substr(k_v_pair.find(':') + 1);
```
Comment (maefromm) [inline truck_inspection_plugin/src/truck_inspection_plugin.cpp:73]: You assume here it's always the first. Can you at least do ".find('VIN:'), will that work?

File: truck_inspection_plugin/src/truck_inspection_plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        resp.success = true;
+    }
+
+    void TruckInspectionPlugin::mobilityOperationCallback(const cav_msgs::MobilityOperationConstPtr& msg)
+    {
+        // if there is a truck around running CARMA
+        if(msg->strategy == TruckInspectionPlugin::INSPECTION_STRATEGY)
+        {
+            // if the incoming message contains a valid safety log
+            if(isSafetyLogValid(msg->strategy_params))
```
Comment (maefromm) [inline truck_inspection_plugin/src/truck_inspection_plugin.cpp:66]: Do you think it'll be wise that we have the check if # of entries=what we expect? Or leave it be?

**Comment by maefromm** [comment]
@qswawrq as discussed with @TonysCousin : 

1. All of the headers of the topics you created for carma-platform and carma-messenger topics (if they have it), should have a timestamp.
2. The CARMAPlatform mobilityoperation topics shall have the header's sender_ bsm_id.
3. The remaining fields on the headers can be left blank.
4. With the timestamp on the header, the CarmaPlatform truck inspection client does not need to have it on the strategy_params. However carma-messenger truck_safety_info will still need that field, so carma-messenger truck inspection plugin will need to digest it and add it back.

Let me know if I had missed anything. (edited)

**Pull request merged**